File: stripe/src/test/java/com/stripe/android/FakeUidSupplier.java
package com.stripe.android;

import android.support.annotation.NonNull;

import java.util.UUID;

final class FakeUidSupplier implements Supplier<StripeUid> {
    @NonNull private final String mValue;

    FakeUidSupplier() {
        this(UUID.randomUUID().toString());
    }

    FakeUidSupplier(@NonNull String mValue) {
        this.mValue = mValue;
    }

    @NonNull
    @Override
    public StripeUid get() {
        return StripeUid.create(mValue);
    }
}

File: stripe/src/test/java/com/stripe/android/FakeUidSupplier.kt
package com.stripe.android

import java.util.UUID

internal class FakeUidSupplier @JvmOverloads constructor(
    private val value: String = UUID.randomUUID().toString()
) : Supplier<StripeUid> {

    override fun get(): StripeUid {
        return StripeUid(value)
    }
}
